data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaKeyMode.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Map;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public enum KafkaKeyMode {
+    DISCARD("discard"),
+    INCLUDE_AS_FIELD("include_as_field"),
+    INCLUDE_AS_METADATA("include_as_metadata");
```

Comment:
Let's be sure to document these new options. I don't see an update the README here. This can be done in a follow PR.

Reply:
Yes. A lot of info is missing in the README. Will update it in a different PR.

Reply:
What does this option would mean for end customer who does not maintain opensource code. how can they use "include_as_metadata"